File: Project/src/plp/expressions2/expression/ValorInteiro.java
package plp.expressions2.expression;

import plp.expressions1.util.Tipo;
import plp.expressions2.memory.AmbienteCompilacao;

/**
 * Objetos desta classe encapsulam valor inteiro.
 */
public class ValorInteiro extends ValorConcreto<Integer> {

	/**
	 * Cria <code>ValorInteiro</code> contendo o valor fornecido.
	 */
	public ValorInteiro(int valor) {
		super(valor);
	}

	
		
	/**
	 * Retorna os tipos possiveis desta expressao.
	 *
	 * @param amb o ambiente de compilação.
	 * @return os tipos possiveis desta expressao.
	 */
	public Tipo getTipo(AmbienteCompilacao amb) {
		return Tipo.TIPO_INTEIRO;
	}
	
	
}

File: Le1 - Python/le1/expression/valor_inteiro.py
from __future__ import annotations

from abc import ABC
from typing import Generic, TypeVar
from le1.expression.valor import Valor
from le1.expression.valor_concreto import ValorConcreto
from le1.expression.expressao import Expressao
from le1.util.tipo import Tipo

T = TypeVar("T")

class ValorInteiro(ValorConcreto[int]):
    def __init__(self, valor: int) -> None:
        super().__init__(valor)

    def getTipo(self) -> Tipo:
        return Tipo.TIPO_INTEIRO

